components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/SystemScopesApiServiceImpl.java

```diff
@@ -71,4 +80,47 @@ public Response systemScopesGet(MessageContext messageContext) throws APIManagem
         }
         return null;
     }
+
+    public Response updateRolesForScope(ScopeListDTO body, MessageContext messageContext) throws APIManagementException {
+        String tenantDomain = MultitenantUtils.getTenantDomain(RestApiUtil.getLoggedInUsername());
+        //read from tenant-conf.json
+        JsonObject existingTenantConfObject = new JsonObject();
+        try {
+            APIMRegistryService apimRegistryService = new APIMRegistryServiceImpl();
+            String existingTenantConf = apimRegistryService.getConfigRegistryResourceContent(tenantDomain,
+                    APIConstants.API_TENANT_CONF_LOCATION);
+             existingTenantConfObject = new JsonParser().parse(existingTenantConf).getAsJsonObject();
+        } catch (RegistryException | UserStoreException e) {
+            APIUtil.handleException("Couldn't read tenant configuration from tenant registry", e);
+        }
+        try {
+
+            JSONObject responseJson = new JSONObject();
+            JSONArray scopeJson = new JSONArray();
+
+            for(int i = 0; i < body.getList().size(); i++) {
+                JSONObject scopeRoleJson = new JSONObject();
+                scopeRoleJson.put("Name", body.getList().get(i).getName());
+                scopeRoleJson.put("Roles", body.getList().get(i).getRoles());
+                scopeJson.put(scopeRoleJson);
+            }
```

Comment:
Can you create a new class SystemScopeMappingUtils and move this logic there and use it from the REST API Layer. This is the typical flow we follow. We should not merge REST API and business logics and include in the same method

Reply:
Fixed with bb70f77